### ISSUE TITLE
Button not disabled with type Submit

### ISSUE THREAD

**Issue by milanvdmria** [grommet/grommet#1558]
### Expected Behavior
Disabled button

### Actual Behavior
Not disabled

### URL, screen shot, or Codepen exhibiting the issue

https://codepen.io/anon/pen/EvbZaz?&editors=0010

### Steps to Reproduce
 1. 
 2. 
 3. 

### Your Environment
* Grommet version: 1.6.0

**Comment by Thuranel** [comment]
Hi!

Thanks for posting the issue. I played a little bit with Button.js and your problem seem to come from
```
        [`${CLASS_ROOT}--disabled`]:
          !onClick &&
          !adjustedHref &&
          ['reset', 'submit'].indexOf(type) === -1
```

The reason why the Button is that way is to make sure we don't break the behavior of a reset or submit button if there is no onClick (See: #1266). Indeed, a <Button type="submit"> should be valid and not disabled.

IMO, having a disabled form button is not always the best idea ( See [Disabled submit button on form?](https://ux.stackexchange.com/questions/9788/disabled-submit-button-on-form-vs-allow-submit-then-show-errors) [When could a disabled button be used?](https://ux.stackexchange.com/questions/73728/should-form-continue-button-be-disabled-if-validation-is-incomplete) ). Also, I've always been told on an accessibility point of view, it is not the best solution either though I can't find a source right now.

If you absolutely need to have a disabled submit button, you could always do a small hack. See:
`
const handler = undefined;
return (
    <Form onSubmit={handler}>
        <Box style={{display: 'none'}}>
            <Button type="submit"/>
        </Box>
        <Button onClick={handler}/>
    </Form>
)
`

In this hack, you have an invisible button that make sure that if you press Enter while in your form, the handler get called by your form. Meanwhile you have a Button that calls the same handler and this button can actually be disabled. This should work, though I am unsure if this is actually a good solution because it might break accessibility.

Ultimately, the best solution should be to have proper support for a `disabled` prop on `<Button/>`s or that if the onSubmit handler for a Form is undefined, the submit button should be disabled too.

**Comment by linde12** [comment]
I'm having the same issue.

I'd like to disable my `<Button type='submit' />` when it's been pressed and API calls are pending. I was trying to do something like this: `<Button type='submit' onClick={submitting ? null : handleSubmit} />` but i think `<Button type='submit' disabled={submitting} />` is much more readable(and also works on `type='submit'` without passing a `onClick`.

I'm working on an embedded device where API calls can take seconds to complete, so this is a must for us.

I understand that this is not beautiful: `<Button type='submit' onClick={() => {}} />`
But not being able to disable a submit button is a bigger problem.

My proposal is the `disabled` prop, but to be consistent it would have to be implemented in Anchor(w/ friends) as well. The "quick" fix is.

tl;dr:
Difficult to solve, but definitely a problem that shouldn't be ignored :/

**Comment by karatechops** [comment]
Tagging #1455.
